### PR TITLE
EVG-5466 more thorough query for finding stuck tasks

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: model/host/db.go [hunk 1]
```diff
@@ -426,6 +426,68 @@ func ByExpiringBetween(lowerBound time.Time, upperBound time.Time) db.Q {
 	})
 }
 
+// StateRunningTasks returns tasks documents that are currently run by a host and stale
+func FindStaleRunningTasks(cutoff time.Duration) ([]task.Task, error) {
+	pipeline := []bson.M{}
```
Comment (tychoish) [inline model/host/db.go]: nit: just do this as a literal thing. pipeline is always the same thing, you don't need to have the append functions.

File: model/host/db.go [hunk 2]
```diff
@@ -426,6 +426,68 @@ func ByExpiringBetween(lowerBound time.Time, upperBound time.Time) db.Q {
...
+					task.StatusKey:       evergreen.TaskDispatched,
+					task.DispatchTimeKey: bson.M{"$lte": time.Now().Add(-2 * cutoff)},
+				},
+			},
+		},
+	})
+
+	tasks := []task.Task{}
+	err := db.Aggregate(Collection, pipeline, &tasks)
+	return tasks, errors.Wrap(err, "error finding stale running tasks")
```
Comment (tychoish) [inline model/host/db.go]: what is tasks if err is not nil?

File: units/task_monitor_execution_timeout.go [hunk 3]
```diff
@@ -49,63 +54,66 @@ func makeTaskExecutionTimeoutMonitorJob() *taskExecutionTimeoutJob {
...
-	flags, err := evergreen.GetServiceFlags()
+	t, err := task.FindOneId(j.Task)
 	if err != nil {
-		j.AddError(err)
+		j.AddError(errors.Wrap(err, "error finding task"))
 		return
 	}
-
-	if flags.MonitorDisabled {
-		grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
```
Comment (tychoish) [inline units/task_monitor_execution_timeout.go]: should keep this, I think

File: model/host/db.go (hunk 1 above)
Comment (john-m-liu) [reply model/host/db.go]: I separated them out for readability. I think this is slightly neater than putting them all in at once

File: model/host/db.go (hunk 2 above)
Comment (john-m-liu) [reply model/host/db.go]: An empty slice. Fixed to return nil

File: units/task_monitor_execution_timeout.go (hunk 3 above)
Comment (john-m-liu) [reply units/task_monitor_execution_timeout.go]: added back in

**Pull request merged**